as/cNFT/assembly/core.ts

```diff
@@ -35,16 +36,42 @@ export function nft_transfer(token_id: TokenId, bidder_id: AccountId): void {
     persistent_tokens.add(token_id, token, bidder_id)
 
     /* Deleting token from previous owner */
-    persistent_tokens.remove(token.id, token.prev_owner_id)
+    persistent_tokens.remove_from_account(token.id, token.prev_owner_id)
+
 
-    
     // Immiting log event
     const transfer_log = new NftTransferLog()
-    
+
     transfer_log.old_owner_id = token.prev_owner_id
     transfer_log.new_owner_id = token.owner_id
     transfer_log.token_ids = [token.id]
 
     const log = new NftEventLogData<NftTransferLog>('nft_transfer', [transfer_log])
     logging.log(log)
+
+}
+
+
+@nearBindgen
+export function burn_design(token_id: TokenId): void {
+
+    /* Getting stored token from tokenId */
+    const token = persistent_tokens.get(token_id)
+
+    assert(context.sender == token.owner_id, "You must be the owner of the token to burn")
```

Comment:
You actually must be `creator_id` to be able to burn. Also I think using predessor id might be better here

Reply:
Used `context.predecessor` for it. But tests are failing. So used `context.sender` as usual.

Reply:
Thanks. 👍